File: src/hyperbolic.cpp
// ------------------------------------------------------------------------
// Copyright (c) 2019-2024 Alexandre Bento Freire. All rights reserved.
// Licensed under the MIT License
// ------------------------------------------------------------------------

// ------------------------------------------------------------------------
//                               Includes
// ------------------------------------------------------------------------

#include "numnjstoix.h"
#include <cfloat>
#include <cmath>
#include <math.h>
#include <node.h>
#include <vector>

namespace numnjstoix {

// ------------------------------------------------------------------------
//                               Actual Functions
// ------------------------------------------------------------------------

/* clang-format off */
transfCalc1DblFunc(sinhFunc, sinh)
transfCalc1DblFunc(coshFunc, cosh)
transfCalc1DblFunc(tanhFunc, tanh)

transfCalc1DblFunc(arcsinhFunc, asinh)
transfCalc1DblFunc(arccoshFunc, acosh)
transfCalc1DblFunc(arctanhFunc, atanh)
/* clang-format on */

// ------------------------------------------------------------------------
//                               Initialize
// ------------------------------------------------------------------------

void initializeHyperbolic(v8exports exports) {

  NODE_SET_METHOD(exports, "sinh", sinhFunc);
  NODE_SET_METHOD(exports, "cosh", coshFunc);
  NODE_SET_METHOD(exports, "tanh", tanhFunc);

  NODE_SET_METHOD(exports, "arcsinh", arcsinhFunc);
  NODE_SET_METHOD(exports, "arccosh", arccoshFunc);
  NODE_SET_METHOD(exports, "arctanh", arctanhFunc);
}

} // namespace numnjstoix

File: test/hyperbolic.js
// ------------------------------------------------------------------------
// Copyright (c) 2019-2024 Alexandre Bento Freire. All rights reserved.
// Licensed under the MIT License
// ------------------------------------------------------------------------

/**
 * Mocha tests for numnjstoix Hyperbolic functions.
 */

// ------------------------------------------------------------------------
//                               Requires
// ------------------------------------------------------------------------

import { numnjstoix } from '../main.js';
import * as shared from './shared.js';

// ------------------------------------------------------------------------
//                               Test Cases
// ------------------------------------------------------------------------

describe('Hyperbolic', function () {

  shared.addTransformCase(describe, 'sinh', Math.sinh, numnjstoix.sinh, true);
  shared.addTransformCase(describe, 'cosh', Math.cosh, numnjstoix.cosh, true);
  shared.addTransformCase(describe, 'tanh', Math.tanh, numnjstoix.tanh, true);

  shared.addTransformCase(describe, 'arcsinh', Math.asinh, numnjstoix.arcsinh, true);
  shared.addTransformCase(describe, 'arccosh', Math.acosh, numnjstoix.arccosh, true);
  shared.addTransformCase(describe, 'arctanh', Math.atanh, numnjstoix.arctanh, true);
});